testcases/crypto/dilithium_func.c

```diff
@@ -240,6 +321,394 @@ CK_RV run_GenerateDilithiumKeyPairSignVerify()
+
+        testcase_begin("Starting Dilithium import key pair, Sign/Verify, KAT index=%lu", i);
+
+        /* Create Dilithium private key */
+        rc = create_DilithiumPrivateKey(session,
+                            dilithium_tv[i].rho, dilithium_tv[i].rho_len,
+                            dilithium_tv[i].seed, dilithium_tv[i].seed_len,
+                            dilithium_tv[i].tr, dilithium_tv[i].tr_len,
+                            dilithium_tv[i].s1, dilithium_tv[i].s1_len,
+                            dilithium_tv[i].s2, dilithium_tv[i].s2_len,
+                            dilithium_tv[i].t0, dilithium_tv[i].t0_len,
+                            dilithium_tv[i].t1, dilithium_tv[i].t1_len,
+                            &priv_key);
+        testcase_new_assertion();
+        if (rc != CKR_OK) {
+            testcase_fail("C_CreateObject (Dilithium Private Key) failed at i=%lu, "
+                          "rc=%s", i, p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+        testcase_pass("*Import Dilithium private key (%s) index=%lu passed.",
+                      dilithium_tv[i].name, i);
+
+        /* Create Dilithium public key */
+        rc = create_DilithiumPublicKey(session,
+                                dilithium_tv[i].rho, dilithium_tv[i].rho_len,
+                                dilithium_tv[i].t1, dilithium_tv[i].t1_len,
+                                &publ_key);
+        testcase_new_assertion();
+        if (rc != CKR_OK) {
+            testcase_fail("C_CreateObject (Dilithium Public Key) failed at i=%lu, "
+                          "rc=%s", i, p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+        testcase_pass("*Import Dilithium public key (%s) index=%lu passed.",
+                      dilithium_tv[i].name, i);
+
+        /* Test sign/verify with KAT */
+        testcase_new_assertion();
+        rc = run_SignVerifyDilithiumKAT(session, i, priv_key, publ_key);
+        if (rc != 0) {
+            testcase_fail("run_SignVerifyDilithiumKAT failed index=%lu.", i);
+            goto testcase_cleanup;
+        }
+        testcase_pass("*Sign & verify KAT, i=%lu passed.", i);
+
+        /* Clean up */
+        rc = funcs->C_DestroyObject(session, publ_key);
+        if (rc != CKR_OK) {
+            testcase_error("C_DestroyObject(), rc=%s.", p11_get_ckr(rc));
+        }
+
+        rc = funcs->C_DestroyObject(session, priv_key);
+        if (rc != CKR_OK) {
+            testcase_error("C_DestroyObject(), rc=%s.", p11_get_ckr(rc));
+        }
+    }
+
+    goto done;
+
+testcase_cleanup:
+    if (publ_key != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, publ_key);
+    if (priv_key != CK_INVALID_HANDLE)
+        funcs->C_DestroyObject(session, priv_key);
+
+done:
+    testcase_user_logout();
+    testcase_close_session();
+
+    return rc;
+}
+
+/**
+ * Creates an AES-256 wrapping key.
+ */
+CK_RV createWrappingKey(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE *secret_key)
```

Comment:
Why not use generate_AESKey from testcases/common/common.c ?

Reply:
Yes, this works. Will replace it. Let's add to our todo list: also clean up this for rsa_func.c and ec_func.c.